src/runners/testngRunner/TestNGRunnerResultAnalyzer.ts

```diff
@@ -83,35 +79,15 @@ export class TestNGRunnerResultAnalyzer implements IRunnerResultAnalyzer {
             }
             this.currentTestState = TestResultState.Failed;
             const testMessages: TestMessage[] = [];
-            if (outputData.attributes.message) {
-                const message: TestMessage = new TestMessage(outputData.attributes.message.trim());
-                if (item.uri && item.range) {
-                    message.location = new Location(item.uri, item.range);
-                }
-                testMessages.push(message);
-            }
 
             if (outputData.attributes.trace) {
-                const traceString: string = outputData.attributes.trace.trim();
                 const markdownTrace: MarkdownString = new MarkdownString();
                 markdownTrace.isTrusted = true;
-                const traceRegExp: RegExp = /(\s?at\s+)([\w$\\.]+\/)?((?:[\w$]+\.)+[<\w$>]+)\(([\w-$]+\.java):(\d+)\)/;
-                for (const line of traceString.split(/\r?\n/)) {
-                    const traceResults: RegExpExecArray | null = traceRegExp.exec(line);
-                    if (traceResults && traceResults.length === 6) {
-                        markdownTrace.appendText(traceResults[1]);
-                        markdownTrace.appendMarkdown(`${(traceResults[2] || '') + traceResults[3]}([${traceResults[4]}:${traceResults[5]}](command:_java.test.openStackTrace?${encodeURIComponent(JSON.stringify([data, this.projectName]))}))`);
-                    } else {
-                        // in case the message contains message like: 'expected: <..> but was: <..>'
-                        markdownTrace.appendText(line.replace(/</g, '&lt;').replace(/>/g, '&gt;'));
-                    }
-                    markdownTrace.appendText('\n');
-                }
                 const testMessage: TestMessage = new TestMessage(markdownTrace);
-                if (item.uri && item.range) {
-                    testMessage.location = new Location(item.uri, item.range);
+
+                for (const line of outputData.attributes.trace.split(/\r?\n/)) {
+                    this.processStackTrace(line, markdownTrace, testMessage, this.currentItem, this.projectName);
```

Comment:
Here `testMessage` is passed as `assertionFailure` but actually it's not.

`assertionFailure` is the test message looks like this:
![image](https://user-images.githubusercontent.com/6193897/130346452-40616b6b-9850-4aef-b0c4-b7090e690c22.png)

you can see it's not a stack trace actually.

I guess that's why the stack trace now only have just few lines.

Reply:
Okay, @jdneo I believe I've resolved the issue with the stack trace that is being presented in the error message. I'm not positive what exactly it is supposed to look like but I've come up with what I believe is the right information.